crates/uv/tests/run.rs

```diff
@@ -114,6 +114,31 @@ fn run_with_python_version() -> Result<()> {
      + sniffio==1.3.1
     "###);
 
+    // This time, we target Python 3.8 instead.
+    let mut command = context.run();
+    let command_with_args = command
+        .arg("--preview")
+        .arg("-p")
+        .arg("3.8")
+        .arg("python")
+        .arg("-B")
+        .arg("main.py")
+        .env_remove("VIRTUAL_ENV");
+
+    uv_snapshot!(context.filters(), command_with_args, @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    Using Python 3.8.[X] interpreter at: [PYTHON-3.8]
+    warning: The Python interpreter (3.8.[X]) is incompatible with the project Python requirement: `>=3.11, <4`
+    Removed virtual environment at: .venv
+    Creating virtualenv at: .venv
+    error: Because the current Python version (3.8.[X]) does not satisfy Python>=3.11,<4 and foo==1.0.0 depends on Python>=3.11,<4, we can conclude that foo==1.0.0 cannot be used.
+    And because only foo==1.0.0 is available and you require foo, we can conclude that the requirements are unsatisfiable.
```

Comment:
Maybe this demonstrates that we should just be erroring early? Because if the interpret doesn't meet the lockfile requirements, we error out.

Reply:
Yeah I saw this elsewhere too we probably shouldn't delete the virtual environment if we're going to fail anyway.

Reply:
Ok, we can change that.